File: Fundamentos de la Robótica/01_Introduccion/05_ode_example/ode_ex.py
import time
import numpy as np
import matplotlib.pyplot as plt
from scipy.integrate import odeint
vo = 10
k1 = 0.5
k2 = 0.35

# Declare the model
def ode_ex_fun(y, t):
   yd0 = vo - k1*y[0]
   yd1 = k1*y[0] - k2*y[1]
   return [yd0, yd1]

   
t = np.linspace(0.0, 20.0, 100)
yinit = np.array([5.0, 0.0])
ti = time.perf_counter()
y = odeint (ode_ex_fun, yinit, t)
tf = time.perf_counter()
print("Elapsed time is %f seconds.\n" %(tf-ti))
plt.plot(t, y[:,0], t, y[:,1]) # y[:,0] is the first column of y
plt.xlabel('t')
plt.ylabel('y')
plt.show()

File: Santiago's Proyects/Fundamentos de la Robótica/01_Introduccion/05_ode_example/ode_ex.m
clear all
close all

p = [10, 0.5, 0.35]
y0 = [0, 0]
tic
[t, y] = ode45(@ode_ex_fun, [0:(20-0)/(100-1):20], y0, [], p)
toc
plot(t,y)
xlabel('t')
ylabel('y')
